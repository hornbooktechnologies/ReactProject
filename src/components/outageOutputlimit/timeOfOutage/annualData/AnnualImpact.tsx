import React from 'react'
import { AnnualImpactModel } from '../../../../models/outageOutputLimit/AnnualImpactModel'
import { useTranslation } from 'react-i18next'
import { cssClassName } from '../../../../utils/CssConstants'
import Tooltip from '../../../common/Tooltip'
import { getDiffOfValues, IsappearTooltip, prefixSign } from '../../../../utils/utils'
import { AppConstant, OutageOutputConstant } from '../../../../utils/AppConstants'

const AnnualImpact = ({
  impactVal,
  impactPreviousVal,
  suffix,
  variation
}: AnnualImpactModel) => {
  const { i18n } = useTranslation()
  let valOfSign = AppConstant.PREFIX
  let diffImpactVal = getDiffOfValues(impactVal,impactPreviousVal)
  
  return (
    <>
      <div className="media-mb-20">
        <div className="uk-flex uk-flex-middle">
          <span className="uk-up-icon uk-text-center uk-inline mr-10">
            <i
              className={`jera uk-position-center font-32 ${variation === OutageOutputConstant.POSITIVE_VAL
                  ? `${cssClassName.JERA_LIKE_ICON_CLASS}`
                  : `${cssClassName.JERA_DISLIKE_ICON_CLASS}`
              }`}
            ></i>
          </span>
          <h6
            className={`uk-margin-remove font-14 ${
              variation === OutageOutputConstant.POSITIVE_VAL
                ? `${cssClassName.COLOR_GREEN_60}`
                : `${cssClassName.COLOR_RED_60}`
            }`}
          >
            {prefixSign(valOfSign,variation)}
            {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.IMPACT')}
            <div>
              <span className={`mt-5 mb-0 uk-text-bold font-14 
                      ${cssClassName.COLOR_WHITE}
                      ${IsappearTooltip(diffImpactVal)} `}
              >
                {prefixSign(impactVal,variation)}
                {impactVal}
                <span className="font-8 uk-text-normal uk-text-capitalize">
                  {suffix}
                </span>
                {diffImpactVal === true ? (
                  <Tooltip
                    prefix={prefixSign(impactPreviousVal,variation)}
                    previousDayValue={impactPreviousVal}
                    unit={suffix}
                  />
                ) : null}
              </span>
            </div>
          </h6>
        </div>
      </div>
    </>
  )
}

export default AnnualImpact
