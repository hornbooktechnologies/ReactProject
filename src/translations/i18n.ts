import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
