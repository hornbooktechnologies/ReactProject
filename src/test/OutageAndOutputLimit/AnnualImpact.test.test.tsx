import React from 'react'
import renderer from 'react-test-renderer'
import AnnualImpact from '../../components/outageOutputlimit/timeOfOutage/annualData/AnnualImpact'
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
describe('Component: AnnualImpact', () => {
  it('AnnualImpact renders correctly', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <AnnualImpact impactVal={10} impactPreviousVal={20} suffix={'oku'} />
        </I18nextProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
