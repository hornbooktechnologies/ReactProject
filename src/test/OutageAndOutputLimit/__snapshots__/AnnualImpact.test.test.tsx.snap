// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AnnualImpact AnnualImpact renders correctly 1`] = `
<div
  className="media-mb-20"
>
  <div
    className="uk-flex uk-flex-middle"
  >
    <span
      className="uk-up-icon uk-text-center uk-inline mr-10"
    >
      <i
        className="jera uk-position-center font-32 color-red-60 jera-dislike"
      />
    </span>
    <h6
      className="uk-margin-remove font-14 color-red-60"
    >
      OUTAGE_AND_OUTPUT_LIMIT.IMPACT
      <div>
        <span
          className="mt-5 mb-0 uk-text-bold font-14 
                      color-white
                      uk-tooltip-bottom-custom uk-tooltip-cutom text-highlight "
        >
          +
          10
          <span
            className="font-8 uk-text-normal uk-text-capitalize"
          >
            oku
          </span>
          <div
            className="uk-transdorm-origin-center-bottom uk-tooltip-bottom-center uk-transform-origin-center-bottom uk-tooltip uk-animation-fade-up uk-animation-center uk-togglabe-enter uk-text-left font-10 uk-text-center font-normal"
          >
            <span
              className="color-grey-blue-80 uk-display-block"
            >
              UNIT_OPERATION.PREVIOUS_DAY
            </span>
            <span
              className="font-10 uk-text-bold uk-display-block mt-5"
            >
              +
              20
              Â 
              oku
            </span>
          </div>
        </span>
      </div>
    </h6>
  </div>
</div>
`;
